src/core/server/base.js

```diff
@@ -269,9 +269,8 @@ function baseServer(routes, createStore, { appInstanceName = appName } = {}) {
 
 export function runServer({ listen = true, app = appName } = {}) {
   if (!app) {
-    log.fatal(
+    throw new Error(
       `Please specify a valid appName from ${config.get('validAppNames')}`);
-    process.exit(1);
```

Comment:
I suppose that might be something that needs to change contextually. Having the process exit when the server is run in production due to a config error is likely to still be desirable.